modules/rpc/rpc.js

```diff
@@ -87,26 +97,112 @@ function mkDir(dirPath, root) {
   return !dirs.length || mkDir(dirs.join(path.sep), root);
 }
 
-let auth = [];
+/*
+** handle RPC call response
+*/
+function cb_handleRequestResponse (res, cb) {
+  var data = '';
+  res.setEncoding('utf8');
+  res.on('data', chunk => {data += chunk});
+  res.on('end', () => {
+    if (res.statusCode === 401) {
+      cb(res);
+      return ;
+    }
+    data = JSON.parse(data);
+    cb(null, data);
+  });
+}
+
+/*
+** execute RPC call
+*/
+function rpcCall (method, params, auth, cb) {
+  var postData = JSON.stringify({
+    method: method,
+    params: params,
+    id: '1'
+  });
+
+  var options = {
```

Comment:
Some of these options could be initialized upfront, for efficiency,
actually all of them except for content-length